src/plugins/unified_search/public/filter_bar/filter_view/index.tsx

```diff
@@ -54,41 +63,59 @@ export const FilterView: FC<Props> = ({
     })} ${title}`;
   }
 
-  const badgeProps: EuiBadgeProps = {
-    title,
-    color: 'hollow',
-    iconType: 'cross',
-    iconSide: 'right',
-    closeButtonProps: {
-      // Removing tab focus on close button because the same option can be obtained through the context menu
-      // Also, we may want to add a `DEL` keyboard press functionality
-      tabIndex: -1,
-    },
-    iconOnClick,
-    iconOnClickAriaLabel: i18n.translate(
-      'unifiedSearch.filter.filterBar.filterItemBadgeIconAriaLabel',
-      {
-        defaultMessage: 'Delete {filter}',
-        values: { filter: innerText },
-      }
-    ),
-    onClick,
-    onClickAriaLabel: i18n.translate('unifiedSearch.filter.filterBar.filterItemBadgeAriaLabel', {
-      defaultMessage: 'Filter actions',
-    }),
-  };
+  // need title to be undefined so that we don't show the native tooltip - we'll use EuiTooltip for the readOnly filter pills instead
+  const readOnlyProps = { title: undefined, color: 'hollow', tabIndex: 0 };
 
-  return (
+  const badgeProps: EuiBadgeProps = readOnly
+    ? readOnlyProps
+    : {
+        ...readOnlyProps,
+        title,
+        iconType: 'cross',
+        iconSide: 'right',
+        closeButtonProps: {
+          // Removing tab focus on close button because the same option can be obtained through the context menu
+          // Also, we may want to add a `DEL` keyboard press functionality
+          tabIndex: -1,
+        },
+        iconOnClick,
+        iconOnClickAriaLabel: i18n.translate(
+          'unifiedSearch.filter.filterBar.filterItemBadgeIconAriaLabel',
+          {
+            defaultMessage: 'Delete {filter}',
+            values: { filter: innerText },
+          }
+        ),
+        onClick,
+        onClickAriaLabel: i18n.translate(
+          'unifiedSearch.filter.filterBar.filterItemBadgeAriaLabel',
+          {
+            defaultMessage: 'Filter actions',
+          }
+        ),
+      };
+
+  const FilterPill = () => (
     <EuiBadge {...badgeProps} {...rest}>
+      <FilterLabel
```

Comment:
If I tried to include the `<span ref={ref}> ... </span>` wrapper here instead, this causes a recursion error on runtime - so had to wrap each `FilterPill` in the `readOnly` ternary expression below 🤷